### DISCUSSION TITLE
Dash license tool with Gradle

### DISCUSSION THREAD

**Discussion by borisrizov-zf** [eclipse/dash-licenses#276] [General]
Our team is working with Gradle and we've written some tasks to ease the use of the dash license tool. I wanted to ask if you're interested in a contribution in that direction. Be it documentation, or maybe the development of a custom Gradle plugin.

The repository is: [tractus-x/managed-identity-wallet](https://github.com/eclipse-tractusx/managed-identity-wallet/tree/main), the file you can view is the `build.gradle` file.

**Comment by borisrizov-zf** [comment]
@AngelikaWittek you asked me to tag you in this.

**Comment by waynebeaton** [comment]
Definitely interested. I can't map the repository; can you give me a URL?

**Reply by borisrizov-zf** [reply]
Sure that would be: https://github.com/eclipse-tractusx/managed-identity-wallet/blob/287b61d12cabf7e2a39f60e1f49d9baaa8401e8c/build.gradle#L161

And I just realised I was linking the wrong repo in the original..

**Reply by waynebeaton** [reply]
It looks good to me. What does a contribution look like?

Do we refactor the dash tasks out of that build file, or does this manifest as some template code for other projects?

How do we invoke this task? (I know next to nothing about Gradle)

**Reply by borisrizov-zf** [reply]
Both is possible. Somewhere down the line this might be converted to a dedicated Gradle plugin, which people can pull in.

I'd suggest we start with a template/gist for other projects. I can pull the code from the project and add it to the docs as a PR, how does that sound?

Invoking is simple:
```bash
gradle dashLicenceCheck
```

where gradle can be substituted with the wrapper `gradlew` or `gradlew.bat`

**Reply by waynebeaton** [reply]
Our current practice is to add templates/demonstration content to the README. If you feel that it fits there, then let's explore that. If you feel that it warrants the creation of an entirely separate document or a pointer to a gist, let's do that.

**Reply by borisrizov-zf** [reply]
@waynebeaton Hi, I've started a PR #295. I saw you have both `dev` and `master` branches and some PRs pull directly into master, that's why I've pointed at `master`. Drop a comment if I should switch it over to `dev`.
